### PR TITLE
Refactor usage of exceptions in converters/connectors

### PR DESCRIPTION
**Pull request by AlanRynne**
These are the existing exceptions right now:

- SpeckleException
- ConversionNotSupportedException
- ConversionException
- UnitNotSupportedException
- ValidationException

Connectors now use exclusively `SpeckleException` instead of a mix of that and conversionexceptions.

The only place where `ConversionNotSupported` is expected is within a `RootObjectConverter`.

`ValidationException` is thrown if we can verify the data is wrong BEFORE we can attempt conversion
`ConversionException` should be used for anything else.

We can expand the types more as we find more use-cases to unify.

### CODE REVIEW

File: Converters/Civil3d/Speckle.Converters.Civil3dShared/Civil3dRootToHostConverter.cs [hunk 1]
```diff
@@ -24,44 +25,23 @@ public Base Convert(object target)
...
+      throw new ConversionNotSupportedException(
         $"Conversion of {target.GetType().Name} to Speckle is not supported. Only objects that inherit from DBObject are."
       );
     }
 
     Type type = dbObject.GetType();
-    object objectToConvert = dbObject;
 
-    // check first for civil type objects
-    if (target is CDB.Entity civilEntity)
```
Comment (adamhathcock) [inline Converters/Civil3d/Speckle.Converters.Civil3dShared/Civil3dRootToHostConverter.cs:36]: intentional?

File: Converters/Revit/Speckle.Converters.RevitShared/Helpers/ParameterValueSetter.cs [hunk 2]
```diff
@@ -108,14 +110,18 @@ private void SetParam(DB.Parameter rp, object value, string units = "", string a
         {
           unitTypeId = new(applicationUnit);
         }
-        else if (scalingService.UnitsToNative(units) is DB.ForgeTypeId typeId)
-        {
-          unitTypeId = typeId;
-        }
         else
```
Comment (adamhathcock) [inline Converters/Revit/Speckle.Converters.RevitShared/Helpers/ParameterValueSetter.cs:113]: I think this fine?

File: Converters/Revit/Speckle.Converters.RevitShared/RevitRootToHostConverter.cs [hunk 3]
```diff
@@ -24,11 +25,6 @@ public object Convert(Base target)
   {
     List<DB.GeometryObject> geometryObjects = _baseToGeometryConverter.Convert(target);
 
-    if (geometryObjects.Count == 0)
```
Comment (adamhathcock) [inline Converters/Revit/Speckle.Converters.RevitShared/RevitRootToHostConverter.cs:27]: is removing this okay?  Should we check for Any?

File: Directory.Build.props [hunk 4]
```diff
@@ -10,6 +10,7 @@
     <CentralPackageVersionOverrideEnabled>true</CentralPackageVersionOverrideEnabled>
     <EnableWindowsTargeting>true</EnableWindowsTargeting> <!-- windows packs for .net core  -->
     <IncludeSourceRevisionInInformationalVersion>false</IncludeSourceRevisionInInformationalVersion>
+    <AutomaticallyUseReferenceAssemblyPackages>false</AutomaticallyUseReferenceAssemblyPackages>
```
Comment (adamhathcock) [inline Directory.Build.props:13]: I think we intentionally removed this?

File: Sdk/Speckle.Converters.Common/Registration/ConverterManager.cs [hunk 5]
```diff
@@ -1,33 +1,35 @@
...
 namespace Speckle.Converters.Common.Registration;
 
-[GenerateAutoInterface]
 public class ConverterManager<T>(ConcurrentDictionary<string, Type> converterTypes, IServiceProvider serviceProvider)
   : IConverterManager<T>
 {
   public string Name => typeof(T).Name;
 
-  public T? ResolveConverter(Type type, bool recursive = false)
+  public T ResolveConverter(Type type, bool recursive = false)
```
Comment (adamhathcock) [inline Sdk/Speckle.Converters.Common/Registration/ConverterManager.cs:12]: Righteous

**Review by adamhathcock — CHANGES_REQUESTED** [review]
I think I follow everything...just making sure about some comments

**Comment by KatKatKateryna** [comment]
Tested in ArcGIS:

- [ ] SpeckleException
- [X] ConversionNotSupportedException
- [X] ConversionException
- [X] UnitNotSupportedException
- [ ] ValidationException

File: Converters/ArcGIS/Speckle.Converters.ArcGIS3/Utils/FeatureClassUtils.cs [hunk 6]
```diff
@@ -311,7 +313,7 @@ List<ObjectConversionTracker> listOfTrackers
       {
         if (trackerItem.HostAppGeom is not ACG.Geometry shape)
         {
-          throw new SpeckleConversionException("Feature Class element had no converted geometry");
+          throw new ValidationException("Feature Class element had no converted geometry");
```
Comment (KatKatKateryna) [inline Converters/ArcGIS/Speckle.Converters.ArcGIS3/Utils/FeatureClassUtils.cs:316]: possibly we can replace some of the ArgumentExceptions in other converters to ValidationException too

**Review by JR-Morgan — APPROVED** [review]

**Pull request merged**